govuk-one-login/authentication-stubs
# Aut 3861/add dynamo and state storing

## What

Adds database for the stub, as well as storing an initial item per request (state for an auth code).

We now generate the auth code, rather than using a hard-coded value. Note, however, that this is not wired together to be the auth code that we return on post - this will come in a subsequent commit + PR. The value of getting this in, for now, is to have a shared database definition to work from.

## How to review

1. Code Review
1. Run locally and create the database according to the instructions in the README
1. Make a request to the locally running stub using the README instructions
1. See that an item is stored in your local database
1. Deploy to dev (currently deployed) and see that the database is successfully created and written to on mfa reset journey
